src/visitor.cpp

```diff
@@ -104,3 +121,65 @@ bool detail::visit(const cpp_entity& e, detail::visitor_callback_t cb, void* fun
     DEBUG_UNREACHABLE(detail::assert_handler{});
     return true;
 }
+
+bool detail::visit(const cpp_entity& e, visitor_filter_t fcb, void* filt_functor,
```

Comment:
So, to clarify, the filtered visit function looks at the children of filtered entities, right?

Then basically, it can be implemented by simply wrapping the user callback:

```cpp
visit(entity, [](...) {
  if (filter(...))
     actual_callback(...);
});
```